source/type/pb_converter.cpp

```diff
@@ -60,6 +61,48 @@ static void fill_codes(const NT* nt, const nt::Data& data, PB* pb) {
     }
 }
 
+template<typename PB>
+static void fill_object_from_ptref(PB* pb,
+                                   const navitia::type::Type_e type_e,
+                                   const nt::Data& data,
+                                   std::string& request,
+                                   int depth,
+                                   const pt::ptime& now,
+                                   const pt::time_period& action_period,
+                                   const bool show_codes) {
```

Comment:
At minimum I'd pass a `navitia::type::Type_e` to tell which type we work on (added to `type_e` which tells the type we want to fill) and remove the string `request` as we can build it inside (we always work on `uri`).
That is if we were not on the verge of breaking all this...
So maybe just rename `type_e` to `attribute_to_fill` (one may find something clearer)